pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1249,13 +1249,17 @@ func (dr *DashboardServiceImpl) FindDashboards(ctx context.Context, query *dashb
 		finalResults := make([]dashboards.DashboardSearchProjection, len(response.Hits))
 		// Create a small runtime cache for folders to avoid extra calls to the folder service
 		foldersMap := make(map[string]*folder.Folder)
+		serviceCtx, serviceIdent := identity.WithServiceIdentity(ctx, query.OrgId)
 		for i, hit := range response.Hits {
 			f, ok := foldersMap[hit.Folder]
 			if !ok {
-				f, err = dr.folderService.Get(ctx, &folder.GetFolderQuery{
+				// We can get search result where user don't have access to parents. If that happens this thi
+				// will fail if we call it as the requesting user. To resolve this we call this as the service so we can
+				// garantuee that we can fetch the parent.
+				f, err = dr.folderService.Get(serviceCtx, &folder.GetFolderQuery{
```

Comment:
This don't seem that great to me, for every unique parent folder in search result we will fetch their parent, would it be possible to add `FolderTitle` to search index? Or would that make re-indexing on change way to annoying?

cc @stephaniehingtgen @JohnnyQQQQ

Reply:
@kalleep reindexing would be a pain.  This is why we did it this way.

Would it be possible to send a list to the folder service?

Reply:
I am not totally sure but I will try to figure it out. If we can we just need to make sure we can do that in both storages I guess. 

I will merge this as is and create an issue to investigate it.

Reply:
@chaudyg @kalleep can you [check this pr](https://github.com/grafana/grafana/pull/100154)?

Reply:
@kalleep indeed we could use label set selectors https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement